lib/frontend/get-jira-configuration.js

```diff
@@ -25,25 +28,34 @@ const formatDate = function (date) {
 module.exports = async (req, res) => {
   const jiraHost = req.query.xdm_e;
   const { client } = res.locals;
-
   const subscriptions = await Subscription.getAllForHost(jiraHost);
-  const installations = await Promise.all(subscriptions.map(subscription => getInstallation(client, subscription)));
+  const installations = await Promise.all(subscriptions
+    .map(subscription => getInstallation(client, subscription)));
+
   const connections = installations
     .filter(response => !response.error)
     .map(data => ({
       ...data,
       isGlobalInstall: data.repository_selection === 'all',
       installedAt: formatDate(data.updated_at),
       syncState: data.syncState,
+      repoSyncState: data.repoSyncState,
     }));
+
   const failedConnections = installations.filter(response => response.error);
 
-  res.render('jira-configuration.hbs', {
-    host: req.query.xdm_e,
-    connections,
-    failedConnections,
-    hasConnections: connections.length > 0 || failedConnections.length > 0,
-    APP_URL: process.env.APP_URL,
-    csrfToken: req.csrfToken(),
-  });
+  try {
+    res.render('jira-configuration.hbs', {
+      host: jiraHost,
+      connections,
+      failedConnections,
+      hasConnections: connections.length > 0 || failedConnections.length > 0,
+      APP_URL: process.env.APP_URL,
+      csrfToken: req.csrfToken(),
+    });
+    return 'Jira configuration rendered successfully.';
```

Comment:
Do we render this string somewhere? Would it be better to return something other than a string? Normally I'd expect this function wouldn't return anything and would `throw` in the case of an error, I see the rendering errors are being `catch`'d below, does this improve the user's experience? Without the original `error` object (and instead a string being returned in the case of an error) there's no easy way to branch on what type of error it is, if later we decide we want to do that.

Reply:
Cleaned this up a little and passed the error to `next` so the error page will render in the event where an error occurs.